src/webhook.rs

```diff
@@ -374,19 +377,19 @@ async fn handle_comment(
 			},
 		)?;
 
-	if body.to_lowercase().trim() == AUTO_MERGE_REQUEST.to_lowercase().trim() {
-		// Fetch the pr to get all fields (eg. mergeable).
-		let pr = github_bot
-			.pull_request(owner, &repo_name, number)
-			.await
-			.map_err(|e| {
-				e.map_issue(Some((
-					owner.to_string(),
-					repo_name.to_string(),
-					number,
-				)))
-			})?;
+	// Fetch the pr to get all fields (eg. mergeable).
+	let pr = github_bot
+		.pull_request(owner, &repo_name, number)
+		.await
+		.map_err(|e| {
+			e.map_issue(Some((
+				owner.to_string(),
+				repo_name.to_string(),
+				number,
+			)))
+		})?;
 
+	if body.to_lowercase().trim() == AUTO_MERGE_REQUEST.to_lowercase().trim() {
```

Comment:
Instead of all these `if`, `else if`, ... why not do:

```rust
match body.to_lowercase().trim() {
  "AUTO_MERGE_REQUEST".to_lowercase().trim() => {
    // Do something
  }
}

Reply:
You add this nice String to enum conversion above:

```rust

		// JobStatus is used by the caller to decide what message to post on Github/Matrix.
		let status = match job.status.to_lowercase().trim() {
			"manual" => JobStatus::Started,
			"running" => JobStatus::AlreadyRunning,
			"success" => JobStatus::Finished,
			"failed" => JobStatus::Finished,
			"canceled" => JobStatus::Finished,
			_ => JobStatus::Unknown,
		};
```

Why not have `handle_comment` take an enum instead of the current `body` `String`?

Reply:
Mostly because I wanted to change as little of the existing code as possible. But now that I got the new code working, refactoring this a bit shouldn't be too hard. I'll give it a try.